src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -239,6 +239,10 @@ public void Encode(RlpStream rlpStream, BlockHeader? header, RlpBehaviors rlpBeh
             {
                 rlpStream.Encode(header.BaseFeePerGas);
             }
+            else if ((rlpBehaviors & RlpBehaviors.Simulate) == RlpBehaviors.Simulate)
```

Comment:
Can we not include `RlpBehaviors` here but use a flag in constructor and use a non-default instance of decoder?

Reply:
Also we need to have the logic not to emit this when block is before EIP1559